jgrapht-core/src/main/java/org/jgrapht/alg/KDisjointShortestPaths.java

```diff
@@ -0,0 +1,328 @@
+/*
+ * (C) Copyright 2016-2016, by Assaf Mizrachi and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+
+import org.jgrapht.DirectedGraph;
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.WeightedGraph;
+import org.jgrapht.graph.AsWeightedDirectedGraph;
+import org.jgrapht.graph.GraphWalk;
+
+/**
+ * The algorithm determines the k <em>disjoint</em> shortest simple paths in increasing order of
+ * weight. Weights can be negative (but no negative cycle is allowed). Only directed simple graphs
+ * are allowed. 
+ * <br>
+ * An undirected graph can be transformed to a directed simple graph using the following
+ * transformation:
+ * <br>
+ * For each pair of vertices V, U and an undirected edge {@literal E:V<-->U} with weight W:
+ * <ol>
+ * <li> Create vertices V_in, V_out, U_in and U_out
+ * <li> Create edges:
+ *     <ol>
+ *     <li> {@literal V_out-->U_in}, weight = W;
+ *  <li> {@literal U_out-->V_in}, weight = W;
+ *  <li> {@literal V_in-->V_out}, weight = 0;
+ *  <li> {@literal U_in-->U_out}, weight = 0;
+ *  </ol>
+ * </ol>
+ * <p>
+ * The algorithm is based on the Suurballe (later extended by Bhandari) algorithm, so is to find an

```

Comment:
Suurballe's first version found a pair of disjoint paths. He later extended it (joint work with Tarjan) to provide for K pairs of disjoint paths. Bhandari later contributed a simpler approach for the applied graph modification after each iteration (in which a simple shortest path is found). Instead of removing the edges of the last shortest path found and modify the cost of all the remained edges in the graph, it is negating them and setting negative cost (only to the shortest path edges) The implementation is clearly taking Bhandari approach. So, bottom line this is BhandriKDisjointShortestsPaths algorithm but I thought I should mention Suurballe and Tarjan. Please advise.

Reply:
I renamed it to BhandariKShortestPaths.